tests/unit/test_sandbox.py

```diff
@@ -108,8 +104,46 @@ def test_ssh_box_failed_cmd_run_as_devin(temp_dir):
         },
         clear=True
     ):
-        ssh_box = DockerSSHBox()
+        for box in [DockerSSHBox(), DockerExecBox()]:
+            exit_code, output = box.execute('non_existing_command')
+            assert exit_code != 0, 'The exit code should not be 0 for a failed command for ' + box.__class__.__name__
 
-        # test the ssh box with a command that fails
-        exit_code, output = ssh_box.execute('non_existing_command')
-        assert exit_code != 0, 'The exit code should not be 0 for a failed command.'
+def test_single_multiline_command(temp_dir):
+    with patch.dict(
+        config.config,
+        {
+            config.ConfigType.WORKSPACE_BASE: temp_dir,
+            config.ConfigType.RUN_AS_DEVIN: 'true',
+            config.ConfigType.SANDBOX_TYPE: 'ssh',
+        },
+        clear=True
+    ):
+        for box in [DockerSSHBox(), DockerExecBox()]:
+            exit_code, output = box.execute('echo \\\n -e "foo"')
```

Comment:
do we expect the agent would pass in `\\\n` when it really want to emit a newline?

Reply:
So the escaping is hard to read here 😄 

This emulates what we'd see if the agent wanted to run
```
echo \
  -e "foo"
```

When writing that out in python, you have to escape the first backslash.